### PR TITLE
allow newer phpstan

### PR DESCRIPTION
**Pull request by juliangut**
PHPStan 0.12 has just been released

### CODE REVIEW

File: src/CompactErrorFormatter.php [hunk 1]
```diff
@@ -26,24 +26,24 @@ public function __construct(
 
 	public function formatErrors(
 		AnalysisResult $analysisResult,
-		OutputStyle $style
+		Output $output
```
Comment (enumag) [inline src/CompactErrorFormatter.php:29]: Pretty sure this won't work with PHPStan 0.10 and 0.11. Change the composer constraint to `^0.12`.

**Review by enumag — CHANGES_REQUESTED** [review]

File: src/CompactErrorFormatter.php (hunk 1 above)
Comment (snapshotpl) [reply src/CompactErrorFormatter.php:29]: So improve tests by running any phpstan version to prove it!

Comment (juliangut) [reply src/CompactErrorFormatter.php:29]: @enumag is right, that interface was introduced in 0.12, I've updated dependencies accordingly

**Review by enumag — APPROVED** [review]

**Comment by enumag** [comment]
ping @jkuchar

File: composer.json [hunk 2]
```diff
@@ -6,8 +6,8 @@
 		"test": "vendor/bin/phpstan analyse -c tests/phpstan-config.neon --error-format oneline --no-progress --no-interaction"
 	},
     "require": {
-        "php": ">= 7.0.0",
-        "phpstan/phpstan": "^0.10 || ^0.11"
+        "php": "^7.1",
+        "phpstan/phpstan": "~0.12"
```
Comment (jkuchar) [inline composer.json]: ^0.12 please ;)

**Comment by jkuchar** [comment]
Thanks!

**Comment by jkuchar** [comment]
released in https://github.com/grifart/phpstan-oneline/releases/tag/v0.3.0-beta1

please let me know if it breaks anything

**Comment by juliangut** [comment]
I'm waiting for https://github.com/ekino/phpstan-banned-code/pull/21 to be merged as well before testing

**Comment by enumag** [comment]
@juliangut ooh thanks for mentioning that repo, didn't know about it. I'll have to add that to my project once it's updated. :-D

@jkuchar I'll try to test the beta on monday.

**Comment by juliangut** [comment]
I've had the time to test 0.3.0 beta1 and it looks like it's working like a charm

**Comment by jkuchar** [comment]
❤️ Here goes the final version! https://github.com/grifart/phpstan-oneline/releases/tag/v0.3.0

**Pull request merged**